receiver/prometheusreceiver/internal/logger.go

```diff
@@ -32,14 +36,89 @@ type zapToGokitLogAdapter struct {
 }
 
 func (w *zapToGokitLogAdapter) Log(keyvals ...interface{}) error {
+	// expecting key value pairs, the number of items need to be even
 	if len(keyvals)%2 == 0 {
-		// expecting key value pairs, the number of items need to be even
-		w.l.Infow("", keyvals...)
+		var lvl zapcore.Level
+		lvl, keyvals = extractLogLevel(keyvals)
+		logFunc, err := levelToFunc(w.l, lvl)
+		if err != nil {
+			return err
+		}
+		logFunc("", keyvals...)
 	} else {
 		// in case something goes wrong
 		w.l.Info(keyvals...)
 	}
 	return nil
 }
 
+// extract go-kit log level from key value pairs, convert it to zap log level
+// and remove from the list to avoid duplication in log message
```

Comment:
Is it idiomatic to start the comment with the function name even for non-exported functions? I've tended to do that anyway, but maybe it's not important